mlflow/entities/_mlflow_object.py

```diff
@@ -25,6 +25,9 @@ def from_dictionary(cls, the_dict):
     def __repr__(self):
         return to_string(self)
 
+    def __eq__(self, other):
```

Comment:
Apparently we also need to implement `__ne__` for inequality and `__hash__` (otherwise this change makes all MLflow entities unhashable), as described in https://stackoverflow.com/a/1227325 - leaving this comment in case anybody looks at this PR or I end up returning to it